state/open.go

```diff
@@ -26,35 +26,76 @@ import (
 	"github.com/juju/juju/worker"
 )
 
-// Open connects to the server described by the given
-// info, waits for it to be initialized, and returns a new State
-// representing the model connected to.
-//
-// A policy may be provided, which will be used to validate and
-// modify behaviour of certain operations in state. A nil policy
-// may be provided.
+// OpenParams contains the parameters for opening the state database.
+type OpenParams struct {
+	// Clock is the clock used for time-related operations.
+	Clock clock.Clock
+
+	// ControllerTag is the tag of the controller.
+	ControllerTag names.ControllerTag
+
+	// ControllerModelTag is the tag of the controller model.
+	ControllerModelTag names.ModelTag
+
+	// MongoInfo is the mongo.MongoInfo used for dialling the
+	// Mongo connection.
+	MongoInfo *mongo.MongoInfo
+
+	// MongoDialOpts is the mongo.DialOpts used to control how
+	// Mongo connections are made.
+	MongoDialOpts mongo.DialOpts
```

Comment:
Please either comment what would happen if dialopts are nil or add to validation to ensure that it does not happen.